src/node/internal/crypto_scrypt.ts

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+//
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* todo: the following is adopted code, enabling linting one day */
+/* eslint-disable */
+
+'use strict';
+
+import { default as cryptoImpl } from 'node-internal:crypto';
+
+import {
+  validateFunction,
+  validateInteger,
+  validateUint32,
+} from 'node-internal:validators';
+
+import {
+  Buffer,
+} from 'node-internal:internal_buffer';
+
+type ArrayLike = cryptoImpl.ArrayLike;
+
+import {
+  kMaxLength,
+} from 'node-internal:internal_buffer';
+
+import {
+  validateByteSource,
+} from 'node-internal:crypto_util';
+
+import {
+  ERR_INVALID_ARG_VALUE,
+} from 'node-internal:internal_errors';
+
+export interface ValidatedScryptOptions {
+  password: ArrayLike;
+  salt: ArrayLike;
+  keylen: number
+  N: number;
+  r: number;
+  p: number;
+  maxmem: number;
+};
+
+export interface ScryptOptions {
+  N?: number;
+  r?: number;
+  p?: number;
+  maxmem?: number;
+  cost?: number;
+  blockSize?: number;
+  parallelization?: number;
+};
+
+const defaults: ScryptOptions = {
+  N: 16384,
+  r: 8,
+  p: 1,
+  maxmem: 32 << 20,
+};
+
+function validateParameters(password: ArrayLike, salt: ArrayLike,
+                            keylen: number, options: ScryptOptions) : ValidatedScryptOptions {
+  // TODO(soon): Add support for KeyObject input.
+  password = validateByteSource(password, 'password');
```

Comment:
Node uses getArrayBufferOrView for password and salt instead (not sure if it makes a difference).